manifests/alertmanager.pp

```diff
@@ -184,13 +190,14 @@
   }
 
   file { $config_file:
-    ensure  => present,
-    owner   => $user,
-    group   => $group,
-    mode    => $config_mode,
-    content => template('prometheus/alertmanager.yaml.erb'),
-    notify  => $notify_service,
-    require => File[$config_dir],
+    ensure       => present,
+    owner        => $user,
+    group        => $group,
+    mode         => $config_mode,
+    content      => template('prometheus/alertmanager.yaml.erb'),
+    notify       => $notify_service,
+    require      => File["${bin_dir}/amtool", $config_dir],
+    validate_cmd => "${bin_dir}/amtool check-config --alertmanager.url='' %",
```

Comment:
do you know if this works for different versions of the alertmanager? Can you provide an acceptance test for this?

Reply:
Doing a quick check on Alertmanager Github page, the amtool command was implemented in version 0.10.0. I could implement a "versioncmp" to make sure to do the validation check if the version is equal or grater than 0.10.0. This works for you?

Reply:
sounds good!